symphony-bdk-core/src/main/java/com/symphony/bdk/core/activity/parsing/InputTokenizer.java

```diff
@@ -0,0 +1,179 @@
+package com.symphony.bdk.core.activity.parsing;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import com.symphony.bdk.core.service.message.util.EntityTypeEnum;
+import com.symphony.bdk.gen.api.model.V4Message;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apiguardian.api.API;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+/**
+ * Class responsible for parsing a {@link V4Message} into a list of tokens separated by at least one whitespace character.
+ * Tokens can be of type {@link String}, {@link Mention}, {@link Cashtag} or {@link Hashtag} depending on the message content.
+ */
+@API(status = API.Status.INTERNAL)
+@Slf4j
+public class InputTokenizer {
+
+  private static final DocumentBuilder DOCUMENT_BUILDER = initBuilder();
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  private static final String DATA_ENTITY_ID = "data-entity-id";
+  private static final String TYPE = "type";
+  private static final String SPAN = "span";
+  private static final String CLASS = "class";
+  private static final String ENTITY = "entity";
+  private static final String ID = "id";
+  private static final String VALUE = "value";
+
+  private static final String SYMPHONY_USER_ID_TYPE = "com.symphony.user.userId";
+  private static final String CASHTAG_VALUE_TYPE = "org.symphonyoss.fin.security.id.ticker";
+  private static final String HASHTAG_VALUE_TYPE = "org.symphonyoss.taxonomy.hashtag";
+
+  @SneakyThrows
+  private static DocumentBuilder initBuilder() {
+    final DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+    return factory.newDocumentBuilder();
+  }
+
+  private Document document;
+  private JsonNode dataNode;
+  private List<Object> tokens;
+  private StringBuffer buffer;
```

Comment:
sonarlint is telling me
_SonarLint: Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder"._ 

(I highly suggest using the plugin in Intellij :))